lib/main.dart

```diff
@@ -48,75 +58,85 @@ class MyHomePage extends StatefulWidget {
-          text: _currentIndex == 0 ? 'Top Stories' : 'New Stories',
+          text: tabs[_currentIndex].name,
           index: _currentIndex,
         ),
-        leading: LoadingInfo(),
+        leading: Consumer<LoadingTabsCount>(
+            builder: (context, loading, child) => LoadingInfo(loading)),
         elevation: 0.0,
-//        actions: [
-//          IconButton(
-//            icon: Icon(Icons.search),
-//            onPressed: () async {
-//              var result = await showSearch(
-//                context: context,
-//                delegate: ArticleSearch(_currentIndex == 0
-//                    ? Provider.of<HackerNewsNotifier>(context).topArticles
-//                    : Provider.of<HackerNewsNotifier>(context).newArticles),
-//              );
-//              if (result != null) {
-//                Navigator.push(
-//                    context,
-//                    MaterialPageRoute(
-//                        builder: (context) => HackerNewsWebPage(result.url)));
-//              }
-//            },
-//          ),
-//        ],
+        actions: [
+          IconButton(
+            icon: Icon(Icons.search),
+            onPressed: () async {
+              var result = await showSearch(
+                context: context,
+                delegate: ArticleSearch(hn.allArticles),
+              );
+              if (result != null) {
+                Navigator.push(
+                    context,
+                    MaterialPageRoute(
+                        builder: (context) => HackerNewsWebPage(result.url)));
+              }
+            },
+          ),
+        ],
       ),
-      body: Consumer<HackerNewsNotifier>(
-        builder: (context, bloc, child) => ListView(
-              key: PageStorageKey(_currentIndex),
-              children: bloc.articles
-                  .map((a) => _Item(
-                        article: a,
-                        prefsBloc: Provider.of<PrefsBloc>(context),
-                      ))
-                  .toList(),
+      body: PageView.builder(
+        controller: _pageController,
+        itemCount: tabs.length,
+        itemBuilder: (context, index) => ChangeNotifierProvider.value(
+              notifier: tabs[index],
+              child: _TabPage(index),
             ),
       ),
-      floatingActionButton: Consumer<bool>(
-        builder: (context, isLoading, child) => FloatingActionButton(
-            onPressed: () {},
-            child: Icon(Icons.shop,
-                color: isLoading ? Colors.yellowAccent : Colors.green)),
-      ),
       bottomNavigationBar: BottomNavigationBar(
         currentIndex: _currentIndex,
         items: [
-          BottomNavigationBarItem(
-            title: Text('Top Stories'),
-            icon: Icon(Icons.arrow_drop_up),
-          ),
-          BottomNavigationBarItem(
-            title: Text('New Stories'),
-            icon: Icon(Icons.new_releases),
-          ),
+          for (final tab in tabs)
```

Comment:
I wonder if just:

`items: tabs.map((tab) => BottomNavigationBarItem(...)).toList()` is cleaner?

Just because this items list consists of exclusively the result of this for-loop. But I am certainly not the style person. Could be persuaded either way.